lib/ui/proposals/details/components/proposal_details_view.dart

```diff
@@ -62,312 +56,365 @@ class _ProposalDetailsViewState extends State<ProposalDetailsView> {
-                  Padding(
-                    padding: const EdgeInsets.symmetric(vertical: 20),
-                    child: ProposalExpirationTimer(
-                      _proposalDetailsModel.formatExpiration(),
-                    ),
-                  ),
-                  const HyphaDivider(),
-                  /// Vote Section
-                  Padding(
-                    padding: const EdgeInsets.symmetric(vertical: 20),
-                    child: Text(
-                      'Cast your Vote',
-                      style: context.hyphaTextTheme.smallTitles,
-                    ),
-                  ),
-                  ...List.generate(
-                    3,
-                        (index) => Padding(
-                      padding: const EdgeInsets.symmetric(vertical: 10),
-                      child: HyphaAppButton(
-                        title: index == 0
-                            ? 'Yes'
-                            : index == 1
-                            ? 'Abstain'
-                            : 'No',
-                        onPressed: () async {},
-                        buttonType: ButtonType.danger,
-                        buttonColor: index == 0
-                            ? HyphaColors.success
-                            : index == 1
-                            ? HyphaColors.lightBlack
-                            : HyphaColors.error,
+                    ...List.generate(
+                      3,
+                          (index) => Padding(
+                        padding: const EdgeInsets.symmetric(vertical: 10),
+                        child: HyphaAppButton(
+                          title: index == 0
+                              ? 'Yes'
+                              : index == 1
+                              ? 'Abstain'
+                              : 'No',
+                          onPressed: () async {},
+                          buttonType: ButtonType.danger,
+                          buttonColor: index == 0
+                              ? HyphaColors.success
+                              : index == 1
+                              ? HyphaColors.lightBlack
+                              : HyphaColors.error,
 
+                        ),
                       ),
                     ),
-                  ),
-                  const SizedBox(height: 20),
-                ],
-              ),
-            );
-          });
-        },
+                    const SizedBox(height: 20),
+                  ],
+                ),
+              );
+            });
+          },
+        ),
       ),
+    );
+  }
+}
+
+Widget _buildTokenRow(
+    BuildContext context,
+    ProposalDetailsModel proposalDetailsModel,
+    int tokenIndex,
+    String tokenType,
+    bool? isShown) {
+  String? amount;
+
+  // Determine the amount and amountPerPeriod based on token type
+  switch (tokenType) {
+    case 'Utility Token':
+      amount = proposalDetailsModel.tokenValue(0, isShown!);
```

Comment:
tokenValue method should use an enum parameter, also _buildTokenRown should use the same enum